h/templates/layouts/base.html

```diff
@@ -10,7 +10,13 @@
       {% endfor -%}
     {% endblock %}
 
-    <title>{% block title %}Hypothesis{% endblock %}</title>
+    <title>
+      {%- block title -%}
+        {%- block page_title %}{% endblock -%}
+        {%- if self.page_title() %} | {% endif -%}
```

Comment:
You're right `self` refers to the template. It's used to call the same block multiple times in a template.

> If you want to print a block multiple times you can however use the special self variable and call the block with that name[1]

We use it here to check whether a child template has provided any content for the `page_title` block and if so render the pipe character to act as a divider. We need to call the function rather than just use `self.page_title` because the block is always defined and returns a string. We just want to know if that string contains anything.

[1] http://jinja.pocoo.org/docs/dev/templates/#child-template